### PR TITLE
Contract setup

### LINKED ISSUE #2: Setup ticket contract architecture in solidity
Context: The ticket contract and transfer of ownership process must be designed in solidity.

### PR DESCRIPTION
**Pull request by sjason19**
fixes #2

### CODE REVIEW

File: contracts/TicketCrypt.sol [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    string name;                                    // Ticket tier/description (Tier 1, Tier 2 etc..)
+    uint price;                                     // Ticket price
+    uint32 available;                               // Available tickets
+    uint32 sold;                                    // Tickets sold
+    mapping (address => uint) ticketToOwner;        // Maps the tickets to the contract owner
+  }
+
+  Ticket[] public tickets;
+
+  function _createEvent(string _name, uint _price, uint32 _available) internal {
```
Comment (patricklinpl) [inline contracts/TicketCrypt.sol]: @sjason19 should we consider tiered pricing?

File: contracts/TicketTransactions.sol [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      _;
+    }
+
+    function purchaseTicket(uint _ticketId) external payable checkAvailability(_ticketId) {
+      Ticket storage t = tickets[_ticketId];
+      t.ticketToOwner[msg.sender] = msg.value;
+      t.sold++;
+      PurchaseTicket(_ticketId, msg.sender, msg.value);
+    }
+}
```
Comment (patricklinpl) [inline contracts/TicketTransactions.sol]: @sjason19 How about refunds?

File: contracts/TicketCrypt.sol (hunk 1 above)
Comment (sjason19) [reply contracts/TicketCrypt.sol]: I was going to add in the extra functionality later, I just wanted to make sure the contract actually works.

File: contracts/TicketTransactions.sol (hunk 2 above)
Comment (sjason19) [reply contracts/TicketTransactions.sol]: I was also going to implement all of that stuff after getting the core part of the contract working.

**Comment by sjason19** [comment]
@patricklinpl I can file additional tickets for ticket refunds and tiered pricing. I was thinking about adding all the additional stuff later just in case it causes errors. you ok with that?

**Pull request merged**